### PR TITLE
Log warnings for docker config cred helper exceptions

### LINKED ISSUE #1529: jib should warn when docker-credential-gcloud is not found for gcr.io
@etanshaul hit a situation where he was unable to push to a repository on `gcr.io`. Although he had configured docker to authenticate using `gcloud auth configure-docker` (such that his `~/.docker/config.json` had credhelpers registered to `gcloud`), his `PATH` was wrong and `docker-credential-gcloud` was not found by Jib.  Jib silently moved on, and his push failed with an `UNAUTHORIZED`.  Jib should instead warn when a configured credential helper in the `config.js` cannot be found.

As an aside, Jib does warn if `docker-credential-gcr` is not found, regardless of whether it was configured as a credential helper.

### PR DESCRIPTION
**Pull request by TadCordle**
Should fix #1529.

### CODE REVIEW

File: jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java [hunk 1]
```diff
@@ -112,7 +119,13 @@ public DockerConfigCredentialRetriever(String registry, Path dockerConfigFile) {
         } catch (IOException
             | CredentialHelperUnhandledServerUrlException
             | CredentialHelperNotFoundException ex) {
-          // Ignores credential helper retrieval exceptions.
+          // Warns the user that the specified (or inferred) credential helper cannot be used.
+          if (eventDispatcher != null && ex.getMessage() != null) {
+            eventDispatcher.dispatch(LogEvent.info(ex.getMessage()));
```
Comment (chanseokoh) [inline jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java]: I think this should be `.warn()` as mentioned in #1529. This means misconfiguration in `.docker/config.json`: a credential helper binary is configured, but the binary couldn't be found or the helper did not return credentials unexpectedly.

You said

> It looks like we do log about missing inferred/configured credential helpers at the info level already

Where is this being done? Is this related to what @briandealwis said about

> As an aside, Jib does warn if docker-credential-gcr is not found, regardless of whether it was configured as a credential helper.

? I vaguely remember Jib warns if a usual cred helper is missing regardless of whether it is configured in `config.json`.

Comment (TadCordle) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java]: `CredentialRetrieverFactory#inferCredentialHelper()` logs `CredentialHelperNotFoundException`/`CredentialHelperUnhandledServerUrlException` at the info level. Explicitly configured credential helpers (`CredentialRetrieverFactory#dockerCredentialHelper()`) seems to rethrow a related `IOException` as a `CredentialHelperNotFoundException()`. I'm not finding anything related that's logged at the `warn` level, maybe @briandealwis knows.

Comment (briandealwis) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java]: I remember seeing complaints about not finding `docker-credential-gcr` though I can't reproduce.  I wonder if I was using an example using an older version of Jib.

Comment (briandealwis) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java]: I agree with @chanseokoh: I tried pushing to gcr.io but without `docker-credential-gcloud` in my `PATH`, where `gcloud` is configured within my `.docker/config.json` for `gcr.io`.  With this patch as is, I saw no helpful messages.  Changing the `info` to `warn`, I get the more helpful:

```
[INFO] --- jib-maven-plugin:1.0.3-SNAPSHOT:build (default) @ helloworld ---
[INFO] 
[INFO] Containerizing application to gcr.io/gcp-testing-29114/foo...
[INFO]    The system does not have docker-credential-gcloud CLI
```

Comment (TadCordle) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java]: Alright, I originally had it at info to be consistent with the other cred helper warnings, but I'll change it to warn for this case.

Comment (briandealwis) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java]: Odd that `eventDispatcher.dispatch(LogEvent.warn(xxx))` is  logged as `[INFO]`.  Hmm so is `.error`.

Comment (TadCordle) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java]: That shouldn't be the case. Gradle/MavenProjectProperties should setup the loggers to log at the correct level.

Comment (chanseokoh) [reply jib-core/src/main/java/com/google/cloud/tools/jib/registry/credentials/DockerConfigCredentialRetriever.java]: > Alright, I originally had it at info to be consistent with the other cred helper warnings, but I'll change it to warn for this case.

In the other case of `CredentialRetrieverFactory#inferCredentialHelper()`, we blindly attempt running `docker-credential-gcr` if the registry is `gcr.io`, regardless of whether the helper is configured in `.docker/config.json`. It's for the convenience sake, e.g., for people who forgot to configure Docker but has the binary on `PATH` for popular registries like `gcr.io`. So, it makes sense that we don't log at the warn (although I actually think it should do warn on `CredentialHelperUnhandledServerUrlException` to be more helpful while it makes sense to do info on `CredentialHelperNotFoundException`.)

**Comment by briandealwis** [comment]
I must have had a stale artifact as it works fine after a rebuild. 🤷‍♂️

**Pull request merged**